pkg/catalogv2/git/git.go

```diff
@@ -128,17 +128,17 @@ func (g *git) Update(branch string) (string, error) {
 	return g.currentCommit()
 }
 
-// Ensure runs git clone, clean DIRTY contents and fetch the latest commit
-func (g *git) Ensure(commit string) error {
+// Ensure runs git clone, clean DIRTY contents and fetch the latest commit in the given branch
+func (g *git) Ensure(branch string) error {
 	if err := g.clone(""); err != nil {
```

Comment:
Should we pass `branch` to the g.clone("")? since now we are cloning the branch

Reply:
I don't really remember why the first clone is done with (").
But this is the default behavior.

Following the Ensure method. 
1. Try to clone, if repo already present, does not clone only opens, which is our case.
2. Try to reset at given branch, if err != nil than we continue to fetchAndReset which is our case, since we probably don't have the branch. 
```go

``` // Ensure runs git clone, clean DIRTY contents and fetch the latest commit in the given branch
func (g *git) Ensure(branch string) error {
	if err := g.clone(""); err != nil {
		return err
	}

	if err := g.reset(branch); err == nil {
		return nil
	}

	return g.fetchAndReset(branch)
}